python/pyspark/sql/functions/builtin.py

```diff
@@ -262,6 +277,14 @@ def lit(col: Any) -> Column:
                 errorClass="COLUMN_IN_LIST", messageParameters={"func_name": "lit"}
             )
         return array(*[lit(item) for item in col])
+    elif isinstance(col, dict):
```

Comment:
the problem is that `dict` can be mapped to `struct`.

Reply:
you might need a configuration or parameter to control this

Reply:
I have added a param to_struct, if you want to check